comms/dht/src/network_discovery/state_machine.rs

```diff
@@ -196,7 +196,12 @@ impl DhtNetworkDiscovery {
     async fn get_next_event(&mut self, state: &mut State) -> StateEvent {
         use State::{Discovering, Initializing, OnConnect, Ready, Waiting};
         match state {
-            Initializing => self::Initializing::new(&mut self.context).next_event().await,
+            Initializing => {
+                let initial_peer_sync_delay = self.config().network_discovery.initial_peer_sync_delay;
```

Comment:
You dont need to pass in a separate delay. The state can access the config using context